File: 2928.distribute-candies-among-children-i.py
#
# @lc app=leetcode id=2928 lang=python3
#
# [2928] Distribute Candies Among Children I
#

# @lc code=start
class Solution:
    '''
    brute force
    '''
    def distributeCandies(self, n: int, limit: int) -> int:
        result = 0
        for i in range(limit + 1):
            for j in range(limit + 1):
                for k in range(limit + 1):
                    if i + j + k == n:
                        result += 1
        return result
# @lc code=end

File: 2928.distribute-candies-among-children-i.cpp
/*
 * @lc app=leetcode id=2928 lang=cpp
 *
 * [2928] Distribute Candies Among Children I
 */

// @lc code=start
class Solution {
public:
    int distributeCandies(int n, int limit) {
        int result = 0;
        for (int i = 0; i <= limit; i++) {
            for (int j = 0; j <= limit; j++) {
                for (int k = 0; k <= limit; k++) {
                    if (i + j + k == n) {
                        result++;
                    }
                }
            }
        }
        return result;
    }
};

// @lc code=end

